pkg/controller/common/reconciler/reconciler.go

```diff
@@ -156,8 +156,15 @@ func ReconcileResource(params Params) error {
 				return err
 			}
 		}
+		reconciledMeta, err := meta.Accessor(params.Reconciled)
+		if err != nil {
+			return err
+		}
+		// retain the resource version to avoid unconditional updates as a work around for https://github.com/kubernetes-sigs/controller-runtime/pull/926
```

Comment:
I'd argue this is the right decision even without that fake client behavior. When deciding where or not to update (and creating the actual desired resource in `UpdateReconciled`), we are basing those decisions off of the state of the resource at that specific RV, so it makes sense to send that RV with the update and use the concurrency mechanisms k8s provides.